subprojects/docs/src/docs/userguide/authoring-builds/software-product/structuring_software_products.adoc

```diff
@@ -0,0 +1,321 @@
+include::sample[dir="samples/build-organization/structuring-software-projects/kotlin",files="domain-model/settings.gradle.kts[tags=include]"]
+====
+
+Once included, you may create a folder matching the project name and create a `build.gradle(.kts)` file in it to configure that part of the component.
+You can find more information in the chapter about <<multi_project_builds.adoc#,structuring Gradle builds for a single software component>>.
+
+== Assigning types to components
+
+Let's zoom into the _domain-model_ component:
+
+[source.multi-language-sample,groovy]
+----
+└── domain-model              <-- component
+    ├── settings.gradle       <-- define inner structure of component and where to locate other components
+    └── release               <-- project in component
+        └── build.gradle      <-- defines type of the project and its dependencies
+----
+
+[source.multi-language-sample,kotlin]
+----
+└── domain-model              <-- component
+    ├── settings.gradle.kts   <-- define inner structure of component and where to locate other components
+    └── release               <-- project in component
+        └── build.gradle.kts  <-- defines type of the project and its dependencies
+----
+
+Initially, `release/build.gradle(.kts)` is empty.
+The project is of no specific type and does not offer any useful content.
+If we add more files to the `domain-model/release` folder now, for example Java source files, Gradle won't know what to do with these files and just ignore them.
+We need to assign a type to the project to make Gradle aware of the purpose of such files in the project.
+
+In Gradle, you assgn a type to a project by <<plugins.adoc#,applying a plugin>>.
+The simplest thing you can do is applying one of Gradle's core plugins, like `base` or `java-library`.
+
+However, usually you have additional configuration to do in the context of the product you are building.
+For example, if your project should be a "Java Library", it would not only apply the `java-library` plugin but also configure details such as setting the Java version to 11.
+You can add details like that directly in `release/build.gradle(.kts)` but than you would have to repeat them in other components that also contain "Java Library" projects.
+
+Thus, the recommendation is to start using custom project types right away:
+
+.Assign a custom type to a component's project by applying a convention plugin
+====
+include::sample[dir="samples/build-organization/structuring-software-projects/groovy",files="domain-model/release/build.gradle[tags=project-type]"]
+include::sample[dir="samples/build-organization/structuring-software-projects/kotlin",files="domain-model/release/build.gradle.kts[tags=project-type]"]
+====
+
+As stated above, a project type is represented by a plugin in Gradle.
+We thus define our custom project type `com.example.kotlin-library` as a plugin as described in the next section.
```

Comment:
The previous paragraph talks about a Java library, setting Java version and this one suddenly makes it a `kotlin-library`. I'd keep it a `com.example.java-library` here to be consistent.

Reply:
Good point. But I do not want to change the sample. But also want to keep the "domain model" as the main example in this section. 

I expanded this a bit to show both a Java and Kotlin convention plugin snippet.